searx/engines/yacy.py

```diff
@@ -22,20 +22,26 @@
 - :py:obj:`search_mode`
 - :py:obj:`search_type`
 
+The :py:obj:`base_url` has to be set in the engine named `yacy` and is used by
+all yacy engines.
+
```

Comment:
> Are we intentonally using only `https://yacy.searchlab.eu` (defined in the python module) for the images category by default?

Nope .. see doc above .. its implemented in `_base_url()` function from line 115 on.

Reply:
Oh okay. But what happens if somebody wants to use different engines for general and image searches because one might be better than the other for images? (I've no idea if that makes sense though, I don't know how yacy works internally)

And if somebody renames the engines to something else than yacy, it won't work either?

Reply:
> Oh okay. But what happens if somebody wants to use different engines for general and image searches because one might be better than the other for images? (I've no idea if that makes sense though, I don't know how yacy works internally)

With current implementation its not possible .. but can be implemented in a future PR as a new feature.

> And if somebody renames the engines to something else than yacy, it won't work either?

No, it won't work anymore / the documentation is clear at this point "has to be set in the engine named `yacy`" .. I assume nobody will rename the yayc WEB engine, IDK a reason for.

In the end, it is a compromise to reduce the maintenance of settings.yml, albeit with little loss in flexibility / we do similar in the [piped configuration](https://docs.searxng.org/dev/engines/online/piped.html#configuration)